main.py

```diff
@@ -49,54 +49,48 @@ def read_files(args):
     # Read files
     films_lines=films_file.readlines()
     collectors_lines=collectors_file.readlines()
-    collector_id=1000
     director_id=3000
     actor_id=2000
-    actor_set=Set()
-    director_set=Set()
+
     # Firstly, read informations in FILMS file
     for i in xrange(len(films_lines)):
 
         line=films_lines[i].split(' % ')
-        actor_list=(line[3].split(', '))
-        #Add directors of movie into director set
-        director_set.add(line[5])
-        #Add actors of the movie into set
-        for actor in actor_list:
-            actor_set.add(actor)
+
         #Send movie informations to dataset and create a movie
         movie = dataset.Movie(
             ID=line[0],
             title=line[1],
             year=line[2],
             genre=line[4],
             director=line[5],
-            rating=line[6]
+            rating=line[6],
         )
-        #Add movie and into movies set in dataset
+        #Add movie into movies set in dataset
         data.movies.add(movie)
-        #Send informations of every director to Director class in dataset
-    for director in director_set:
+
+        movie.directors=Set(line[5])

```

Comment:
```
data.movies.add(movie)
movie.directors=Set(line[5])
```
bu kısımda movie'yi henüz tamamlamadan data'ya atmış oluyorsun.
Doğrusu:
```
movie.directors=Set(line[5])
movie.actors=Set(...
data.movies.add(movie)
```
şeklinde olmalı

Reply:
Hocam mesela burda Set(line[5]) dediğimizde line5 teki insanın isminin harflerinden set oluşturuyor yani bana directorlerin setini vermiyor. Bende    

movie.directors.add(line[5]) şeklinde tanımladım ve şöyle devam ettim

    #Send informations of every director to Director class in dataset

        for director in movie.directors:

        #Create director

            director=dataset.Director(

                ID=director_id,

                name=director

            )

            director_id+=1

        # Add director into directors set in dataset
            data.directors.add(director)

Şeklinde yaptım fakat main de directorleri print ettiğimde aynı directorlerin yine döndüğünü gördüm.Onu ayarlayamadım

Reply:
Demek istediğim movie içinde director olmadan data'ya atıyorsun. içine director koyduktan sonra dataya atman lazım.

Reply:
movie değişkenini tamamen doldurduktan sonra en son data ya at

Reply:
"Hocam mesela burda Set(line[5]) dediğimizde line5 teki insanın isminin harflerinden set oluşturuyor yani bana directorlerin setini vermiyor. "
Bu kısma takılmadım, neye yorum yaptığıma dikkat :)

ki zaten directorların setini de vermeyebilir sadece isimlerini tutalım moviede. yoksa çok fazla data redundant bir şekilde memoryde yer işgal edecek

Reply:
Files methodunu gönderdim hocam dediğiniz gibi en altta dataya yolladım movieyi. Bu şekilde mi olması gerek :


def read_files(args):

    try:

        # Open files

        films_file=open(args.films_path,"r")

        collectors_file=open(args.collectors_path,"r")

    except IOError:

         #if there is a error, do it

        print("The files are not found.")

        sys.exit()



    # Create data instance

    data = dataset.Data()



    # Read files

    films_lines=films_file.readlines()

    collectors_lines=collectors_file.readlines()

    director_id=3000

    actor_id=2000



    # Firstly, read informations in FILMS file

    for i in xrange(len(films_lines)):

        line=films_lines[i].split(' % ')

        #Send movie informations to dataset and create a movie

        movie = dataset.Movie(

            ID=line[0],

            title=line[1],

            year=line[2],

            genre=line[4],

            director=line[5],

            rating=line[6],

        )

        movie.directors.add(line[5])
        movie.actors=Set(line[3].split(', '))

        #Send informations of every director to Director class in dataset

        for director in movie.directors:
            director=dataset.Director(

                ID=director_id,

                name=director

            )

            director_id+=1

        # Add director into directors set in dataset
            data.directors.add(director)


    #Send every actor to Actor class in dataset

        for actor in movie.actors:
            #Create actor
            actor=dataset.Actor(
                ID=actor_id,
                name=actor
            )
            actor_id+=1

        # Add actor into actors set in datasetc

            data.actors.add(actor)


        #Add movie into movies set in dataset

        data.movies.add(movie)

    #Read lines in collectors file

    for i in xrange(len(collectors_lines)):

        line=collectors_lines[i].split('%')

        #Send every collector to Collector class in dataset

        collector = dataset.Collector(

            ID=line[0],

            name=line[1],

            email=line[2]

        )

        #Add collector into collectors set in data class

        data.collectors.add(collector)

    return data